hypothesis-python/src/hypothesis/reporting.py

```diff
@@ -8,48 +8,57 @@
 # v. 2.0. If a copy of the MPL was not distributed with this file, You can
 # obtain one at https://mozilla.org/MPL/2.0/.
 
+from contextlib import AbstractContextManager
+from typing import TYPE_CHECKING, Callable
+
 from hypothesis._settings import Verbosity, settings
 from hypothesis.internal.compat import escape_unicode_characters
 from hypothesis.utils.dynamicvariables import DynamicVariable
 
+if TYPE_CHECKING:
+    from typing import TypeAlias
+
 
-def default(value):
+def default(value: object) -> None:
     try:
         print(value)
     except UnicodeEncodeError:
+        assert isinstance(value, str)
         print(escape_unicode_characters(value))
```

Comment:
```suggestion
        print(escape_unicode_characters(str(value)))
```
`value` might be a non-string object with a `__str__` method that returns something unprintable, so let's just cast it.